horizomer/utils/tree.py

```diff
@@ -17,6 +17,253 @@
 from skbio.tree import MissingNodeError
 
 
+def _extract_support(node):
+    """Extract the support value from a node label, if available.
+    Returns
+    -------
+    tuple of
+        int, float or None
+            The support value extracted from the node label
+        str or None
+            The node label with the support value stripped
+    """
+    support, label = None, None
+    if node.name:
+        # separate support value from node name by the first colon
+        left, _, right = node.name.partition(':')
+        try:
+            support = int(left)
+        except ValueError:
+            try:
+                support = float(left)
+            except ValueError:
+                pass
+        # strip support value from node name
+        label = right or None if support is not None else node.name
+    return support, label
+
+
+def _node_label(node):
+    """Generate a node label in the format of "support:name" if both exist,
+    or "support" or "name" if either exists.
+    Returns
+    -------
+    str
+        Generated node label
+    """
+    lblst = []
+    if node.support is not None:  # prevents support of NoneType
+        lblst.append(str(node.support))
+    if node.name:  # prevents name of NoneType
+        lblst.append(node.name)
+    return ':'.join(lblst)
+
+
+def assign_supports(tree):
+    """Extract support values from internal node labels of a tree.
+    Notes
```

Comment:
We need some blank lines here. When you copy something directly from GitHub, the blank lines are gone.